### DISCUSSION TITLE
RL based flight control using Simulink and JSBSim S Function.

### DISCUSSION THREAD

**Discussion by sherwinkumar2004** [JSBSim-Team/jsbsim#1100] [Q&A]
Hey everyone, 
    I am trying to implement altitude and heading control using reinforcement learning in Matlab. However I am facing some issues in training as the rewards don't seem to converge. 

![Training 1097 episodes](https://github.com/JSBSim-Team/jsbsim/assets/108203310/683f0ae8-e5d2-4bd6-a191-6c27a8fb4a61)

Here I have trained for 1097 episodes. For the hyperparameters and other training parameters I have followed the research paper by @agodemar on the same. Can you please let me know what could be the possible reasons for such a poor training?

**Comment by agodemar** [comment]
@sherwinkumar2004 you do not give details on what you are doing. What is the aircraft model you are using?

**Reply by sherwinkumar2004** [reply]
I am extremely sorry for not mentioning the details. I am using a F-16 aircraft model from JSBSim. I have used your research paper for guidance regarding the DDPG actor and critic parameters. Here I have set a target of 27,000ft and the initial conditions as 30,000ft.
